src/utils/bitcoin.ts

```diff
@@ -0,0 +1,30 @@
+import { AxiosError } from 'axios';
+
+import { getMarketsBySymbols } from './coingecko';
+import { isDefined } from './helpers';
+import logger from './logger';
+import SingleQueryDataProvider from './SingleQueryDataProvider';
+
+const getBtcExchangeRate = async () => {
```

Comment:
`getTezExchangeRate` and `getBtcExchangeRate` have too much common logic. It would be better to create a `createCoingeckoExchangeRateProvider` method and use it in both cases